Automattic/wp-calypso
# Settings: Cleanup analytics settings from legacy Jetpack version checks

We are supposed to support only Jetpack current version - 1, so we can clean up any older version checks from Calypso. This PR does it for the Google analytics settings.

Part of #29888

#### Changes proposed in this Pull Request

* Cleanup all the Jetpack legacy version checks from Google analytics settings.
* Cleanup any unused props in Google Analytics settings form.

#### Testing instructions

* Checkout this branch, or spin it up on calypso.live.
* Perform the following operations for:
  * A Jetpack site (with Jetpack >= 6.7.0).
  * A WP.com site.
  * An Atomic site.
  * (bonus) a VIP site.
* Go to Traffic settings of the site.
* Verify saving in the Google Analytics card works well just like it did before.
* Compare with production on the same site and verify the same is displayed in the Google Analytics card.

Note: there are some unused selectors that can be deleted after this PR has been merged, but I'll do that in a separate PR (see #29888).
